x-pack/legacy/plugins/apm/public/components/app/Settings/AddSettings/AddSettingFlyout.tsx

```diff
@@ -88,6 +89,18 @@ export function AddSettingsFlyout({
       env.name === environment && (Boolean(selectedConfig) || env.available)
   );
 
+  useEffect(() => {
+    if (selectedConfig) {
+      setEnvironment(selectedConfig.service.environment);
+      setServiceName(selectedConfig.service.name);
+      setSampleRate(selectedConfig.settings.transaction_sample_rate.toString());
+    } else {
+      setEnvironment(ENVIRONMENT_NOT_DEFINED);
+      setServiceName(undefined);
+      setSampleRate('');
+    }
+  }, [selectedConfig]);
```

Comment:
I think you should either use initial state or `useEffect`. Not both. Eg. currently `useState` looks like:
```ts
  const [sampleRate, setSampleRate] = useState<string>(
    selectedConfig
      ? selectedConfig.settings.transaction_sample_rate.toString()
      : ''
  );
```

but the initial state is never used.



In 7.5 I'm solving this by adding a `key` prop to the flyout:
https://github.com/sqren/kibana/blob/fff51cdfedd940233080e975dbadb7bb274c333d/x-pack/legacy/plugins/apm/public/components/app/Settings/AgentConfigurations/index.tsx#L49-L54